### DISCUSSION TITLE
How to specify the nonlinear solver and nonlinear solve options used in IDAS integrator?

### DISCUSSION THREAD

**Discussion by eaglema** [casadi/casadi#3088] [Q&A]
Hi all,

I'm using IDAS to solve a DAE model in CasADi, but the DAE model seems to be very still at a time point and the integrator frequently failed with errors like 

_At t = 96.01 and h = 9.74626e-16, the error test failed repeatedly or with |h| = hmin._
or 
_The linesearch algorithm failed with too small a step._

I doubt this might be because the nonlinear equation solver or the options for the nonlinear equation solver adopted by IDAS are not strong enough to solve the nonlinear equation system at that time point. That causes the integrator keeps reducing the step length until the step length is close to the machine epsilon. 

Therefore, I want to tune the options for the nonlinear equation solver used in IDAS, but I didn't find how to do this in the user reference for IDAS in CasADi. Could anyone give me some hints on this? Thank you in advance.

Best wishes,
Yingjie

**Comment by brunoml5** [comment]
Have you tried changing the options shown in the [Python API](https://web.casadi.org/python-api/#integrator)? Maybe give us an example so that understand what you are trying to do.

Also, take a look at the [IDAS documentation](https://sundials.readthedocs.io/en/latest/idas/index.html).
